src/Maestro/maestro-angular/src/maestro-client/models.ts

```diff
@@ -1007,6 +1008,238 @@ export class BuildRef {
     }
 }
 
+export class FlowGraph {
+    public constructor(
+        {
+            nodes,
+            edges,
+        }: {
+            nodes: FlowRef[],
+            edges: FlowEdge[],
+        }
+    ) {
+        this._nodes = nodes;
+        this._edges = edges;
+    }
+
+    private _nodes: FlowRef[];
+    private _edges: FlowEdge[];
+
+    public get nodes(): FlowRef[] {
+        return this._nodes;
+    }
+
+    public get edges(): FlowEdge[] {
+        return this._edges;
+    }
+    
+    public isValid(): boolean {
+        return (
+            this._nodes !== undefined && this._edges !== undefined
+        );
+    }
+
+    public static fromRawObject(value: any): FlowGraph {
+        let result = new FlowGraph({
+            nodes: value["flowRefs"] == null ? undefined : value["flowRefs"].map((n: any) => FlowRef.fromRawObject(n)) as any,
+            edges: value["flowEdges"] == null ? undefined : value["flowEdges"].map((e: any) => FlowEdge.fromRawObject(e)) as any,
+        });
+        return result;
+    }
+
+    public static toRawObject(value: FlowGraph): any {
```

Comment:
>toRawObject [](start = 18, length = 11)

What are these "rawObjects" and why aren't they typed?

Reply:
This is the api generator code. "RawObject" is "what should i serialize to json"

Reply:
This seems to be in a lot of places. We have MyRef, a bunch of any/RawObjects, and then a bunch of anonymous types with all the same values.

Why are all three of these types floating around the system?  Is one of them the wire representation of things?  If so, why isn't it imported from the generated client? We really shouldn't have any untyped values floating around.

---
In reply to: [373736353](https://github.com/dotnet/arcade-services/pull/924#discussion_r373736353) [](ancestors = 373736353)

Reply:
Didn't realize this was all autogenerated. Ignoring it!